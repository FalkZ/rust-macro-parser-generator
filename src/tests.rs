macro_rules! tests {
    (@debug: $name:ident, $t:ident, $str:literal) => {
        #[test]
        fn $name() {
            let r = crate::m1n::renderer::render($str).expect("couldn't compile file");
            insta::assert_debug_snapshot!(&r.$t);
        }
    };
    (@display: $name:ident, $t:ident, $str:literal) => {
        #[test]
        fn $name() {
            let r = crate::m1n::renderer::render($str).expect("couldn't compile file");
            insta::assert_display_snapshot!(&r.$t);
        }
    };
    // `()` indicates that the macro takes no argument.
    ($($name:ident: $str:literal),*) => {
        mod tokens {
            $(
                tests!(@debug: $name, tokens, $str);
            )*
        }
        mod statements {
            $(
                tests!(@debug: $name, statements, $str);
            )*
        }
        mod typescript {
            $(
                tests!(@display: $name, typescript, $str);
            )*
        }
    };
}

tests!(
    enums: "./examples/m1n/Enum.m1n",
    class: "./examples/m1n/Class.m1n",
    file: "./examples/m1n/file.m1n",
    math: "./examples/m1n/std/math.m1n",
    matcher: "./examples/m1n/match.m1n",
    assignment: "./examples/m1n/assignment.m1n",
    operator: "./examples/m1n/operator.m1n"
);
