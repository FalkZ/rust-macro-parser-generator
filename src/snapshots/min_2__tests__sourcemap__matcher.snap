---
source: src/tests.rs
expression: "&r.sourcemap"
---
import util, { pipe } from '@std/util';
import math from '@std/math';





 const self = {}




function matchSimple(v) { return util['match'](v,(_) => { if (_ ==1){ return 1} else if (_ ==`a string`){ return _} else if (_ ==-2){ return 2}});}

function matchExample(v) { return pipe(v).op(util['match'], (_) => { if (_ ==1){ return 1} else if (_ ==`a string`){ return _} else if (_ ==-2){ return 2}}).op(math['+'], 2).end();}export default { ...self, }


AAAA,AACO,OACH,IAAW,IAAE,YAAb
AAAA,OACA,UAAA
AAAA
AAAA
AAAA
AAAA
AAAA
AAAA
AAAA
AAAA
AAAA
AAAA
AAAA,AAAY,SAIhB,WAAY,CAAA,YAAI,AACX,cAAD,EAAO,YAAE,GAAA,EACH,AAAC,WAAI,AAAC,cAAE,EACP,oBAAa,AAAC,cAAE,EACjB,CAAE,WAAI,AAAC
AAAA
AAAA,AACX,SAEN,YAAa,CAAA,YAAI,KACb,MAAC,eAAM,YAAE,GAAA,EACH,AAAC,WAAI,AAAC,cAAE,EACP,oBAAa,AAAC,cAAE,EACjB,CAAE,WAAI,AAAC,IACZ,IAAA,AACD,WAAC,AAAC
